src/mlpack/methods/neighbor_search/ns_model.hpp

```diff
@@ -13,12 +13,24 @@
 #include <mlpack/core/tree/binary_space_tree.hpp>
 #include <mlpack/core/tree/cover_tree.hpp>
 #include <mlpack/core/tree/rectangle_tree.hpp>
-
+#include <boost/variant.hpp>
```

Comment:
I can see a note about this in boost's documentation: [[ref]](http://www.boost.org/doc/libs/1_61_0/more/getting_started/unix-variants.html#id19)
"**Other Packages**
RedHat, Debian, and other distribution packagers supply Boost library packages, however you may need to adapt these instructions if you use third-party packages, because their creators usually choose to break Boost up into several packages, reorganize the directory structure of the Boost distribution, and/or rename the library binaries. If you have any trouble, we suggest using an official Boost distribution from SourceForge. "

I am not sure. But it looks like all partial packages depends on libboost-dev package, so they will finally include all headers files.
For example:  libboost-program-options-dev    *--depends on-->*   libbooost-dev